Client/Frontend/Browser/Onboarding/OnboardingAdsCountdownViewController.swift

```diff
@@ -0,0 +1,128 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import BraveRewards
+import BraveRewardsUI
+import BraveShared
+
+class OnboardingAdsCountdownViewController: OnboardingViewController, UNUserNotificationCenterDelegate {
+    
+    private var rewards: BraveRewards?
+    private var notificationsHandler: AdsNotificationHandler?
+    private var rewardsObserver: LedgerObserver?
+    private var publisher: PublisherInfo?
+    private var timeSinceAnimationStarted: Date?
+    
+    private var contentView: View {
+        return view as! View // swiftlint:disable:this force_cast
+    }
+    
+    override func loadView() {
+        view = View(theme: theme)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        setupBraveRewards()
+        
+        //Countdown timer
+        contentView.countdownText = "3"
+        
+        //On this screen, when you press "Start Browsing", we need to mark all onboarding as complete, therefore we trigger `skip`..
+        contentView.finishedButton.addTarget(self, action: #selector(skipTapped), for: .touchDown)
+        
+        //On this screen, when you press "I didn't see an ad", we need to go to the next screen..
+        contentView.invalidButton.addTarget(self, action: #selector(continueTapped), for: .touchDown)
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        if let time = timeSinceAnimationStarted, Date().timeIntervalSince(time) >= 3.0 {
+            self.contentView.setState(.adConfirmation)
+            return
+        }
+        
+        timeSinceAnimationStarted = Date()
+        contentView.resetAnimation()
+        contentView.animate(from: 0.0, to: 1.0, duration: 3.0) { [weak self] in
+            guard let self = self else { return }
+            self.displayMyFirstAdIfAvailable {
+                self.skipTapped()
+            }
+        }
+    }
+    
+    override func applyTheme(_ theme: Theme) {
+        styleChildren(theme: theme)
+        contentView.applyTheme(theme)
+    }
+}
+
+extension OnboardingAdsCountdownViewController {
+    
+    private func setupBraveRewards() {
+        #if NO_REWARDS
+        rewards = nil
+        rewardsObserver = nil
+        #else
+        RewardsHelper.configureRewardsLogs()
+        rewards = BraveRewards(configuration: .default)
```

Comment:
Hmmm, none of this is needed at all in onboarding. All we need is 1. to check that BraveAds is supported in this region, which doesn't require us make a new one, and 2. the `AdsViewController.displayFirstAd` call

In fact, the only place we need an actual rewards instance is when clicking "accept" or w/e the button is where we need to create the wallet and enable rewards (which I think this PR is still missing). This should be passed _into_ the onboarding controller since the BVC already makes that object (which you copied here)

Is there a reason you added all this and the rewards observers and everything?

Reply:
It is all there because I need to know if the notification has shown or not. Otherwise I can just force to the next screen I guess.. I'll remove it.

As for the loading. I don't know what you mean. I don't have anything like this in the design or requirements :S

Reply:
The `AdsNotificationHandler` is completely separate from the "My First Ad".  You know it showed up based on whether or not it was dismissed/timed out/was clicked/etc. Also the `notificationAdded` observer method isn't for ads, it's for the wallet notifications (insufficient funds, grant added, etc.)

The loading part, means the whole point of this onboarding is to get users to opt-in to Brave Rewards. To opt-in you have to create a wallet (like the panel does). Simply showing them a first ad is nothing but a special UI.

The spec for this is in Drive: `User Onboarding iOS - Phase 2` under **Cards**, point 3 & 4

Reply:
The code related to creating a wallet is in brave-rewards-ios: https://github.com/brave/brave-rewards-ios/blob/5fe6a61bd8735308bd6f079b21938a038fe22301/BraveRewardsUI/Create%20Wallet/CreateWalletViewController.swift#L62

Reply:
This can't be right. I need to know if the ad shown as well.. not just if the user tapped it.
So that I can show the "Did you see an ad screen" or take them directly to the browser.. but it's fine I'll just remove that and show it always then.

Creating the wallet part is fine.

Reply:
If you need additional stuff from the My First Ad display the code for it is here and could be modified to support more than just tapping:

https://github.com/brave/brave-rewards-ios/blob/5fe6a61bd8735308bd6f079b21938a038fe22301/BraveRewardsUI/Ads/AdsViewController.swift#L308

There's nothing really to notify about "this was shown". We just assume it was shown because we're adding it to the view hierarchy ourselves. Maybe we could add a callback for the "animate out" part and if the user didn't tap it before it disappeared then we show the did you see it part, but just adding more complexity

Reply:
Done. Code is added to create a wallet. It is commented out for now until the interface is public.

Added loader when the user agrees. Removed all the rewards observer code and kept only ad creation code.

Reply:
Okay so make the changes in brave-rewards-ios and update the Carthage dep here (Cartfile.resolved update commited) when it's done and uncomment and will continue review then

Reply:
If you need additional updates for the first ad stuff probably best to do them at the same time first in brave-rewards-ios and then you can fix both when the dep is updated in this PR

Reply:
Done.